### PR TITLE
Ethantr/unavailability model

### PR DESCRIPTION
**Pull request by ethantr**
## Describe your changes
Added a freezed model for volunteer unavailability, which can also transform the json data structure to a unavailability object.
## Issue ticket number and link
FE-48 [](https://fireapp-emergiq-2024.atlassian.net/jira/software/projects/FE/boards/1?selectedIssue=FE-48)

Would also like a comment on "unavailability". Is this an appropriate name, or should it go under calendar?

### CODE REVIEW

File: fireapp/lib/domain/models/unavailability/unavailability_time.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+// This associates 'unavailability_time.dart' with the code generated by Freezed
```
Comment (emilymclean) [inline fireapp/lib/domain/models/unavailability/unavailability_time.dart]: This comment is unnecessary

**Review by emilymclean — CHANGES_REQUESTED** [review]
Remove the comment and, assuming this model is correct, this is fine!

File: fireapp/lib/domain/models/unavailability/unavailability_time.dart (hunk 1 above)
Comment (ethantr) [reply fireapp/lib/domain/models/unavailability/unavailability_time.dart]: Fixed!

**Review by emilymclean — APPROVED** [review]
Looks good to me, though I suspect "status" may need to be updated in the implementation to something more descriptive than a boolean (like an enum)

**Comment by emilymclean** [comment]
As a side note, please name your branch ethantr/<feature, bugfix, hotfix, fix>/whatever

**Comment by ethantr** [comment]
> Looks good to me, though I suspect "status" may need to be updated in the implementation to something more descriptive than a boolean (like an enum)

Fair point, will bring this up with the team as it is in the back end as well. Cheers!

**Pull request merged**